File: cleanup-unused-imports.js
import fs from 'fs';
import { glob } from 'glob';

// Function to fix a single file
function fixFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Remove unused imports
    const lines = content.split('\n');
    const newLines = [];
    const usedImports = new Set();
    
    // Find what's actually used in the component
    const componentContent = content.substring(content.indexOf('function '), content.lastIndexOf('}'));
    
    for (const line of lines) {
      if (line.startsWith('import ')) {
        // Check if this import is used
        const importMatch = line.match(/import\s+(?:{[^}]+}|\w+)\s+from/);
        if (importMatch) {
          const importName = importMatch[0].match(/import\s+{?([^}]+)}?\s+from/)[1];
          const cleanImportName = importName.replace(/\s*{[^}]*}\s*/, '').trim();
          
          if (componentContent.includes(cleanImportName) || 
              cleanImportName === 'React' || 
              cleanImportName === 'ErrorBoundary' ||
              line.includes('ErrorBoundary')) {
            newLines.push(line);
          } else {
            // Skip unused import
            modified = true;
          }
        } else {
          newLines.push(line);
        }
      } else {
        newLines.push(line);
      }
    }
    
    if (modified) {
      content = newLines.join('\n');
    }
    
    // Fix ErrorBoundary import path for specific files
    if (filePath.includes('app/page.tsx') || filePath.includes('app/it-services/')) {
      if (content.includes("import ErrorBoundary from '../components/ErrorBoundary'")) {
        content = content.replace(
          "import ErrorBoundary from '../components/ErrorBoundary'",
          "import ErrorBoundary from './components/ErrorBoundary'"
        );
        modified = true;
      }
    }
    
    if (modified) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${filePath}`);
    }
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
  }
}

// Get all page.tsx files
async function fixAllFiles() {
  try {
    const files = await glob('app/**/page.tsx');
    files.forEach(fixFile);
    console.log('All files processed!');
  } catch (error) {
    console.error('Error:', error);
  }
}

fixAllFiles();

File: cleanup_unused_imports.py
#!/usr/bin/env python3
"""
Script to remove unused imports and variables.
"""

import os
import re

def cleanup_file(file_path):
    """Clean up unused imports in a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        lines = content.split('\n')
        new_lines = []
        
        for line in lines:
            # Remove unused SEOHead imports
            if 'import SEOHead' in line and 'SEOHead' not in content.replace(line, ''):
                continue
            
            # Remove unused ComponentsPage references
            if 'ComponentsPage' in line and 'is declared but its value is never read' in str(content):
                continue
                
            # Remove unused variable declarations
            if 'enableVirtualScrolling' in line and 'is declared but its value is never read' in str(content):
                continue
            if 'enableCodeSplitting' in line and 'is declared but its value is never read' in str(content):
                continue
                
            new_lines.append(line)
        
        new_content = '\n'.join(new_lines)
        
        # Additional cleanup
        new_content = re.sub(r'\n\s*\n\s*\n', '\n\n', new_content)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        
        print(f"Cleaned up: {file_path}")
        return True
        
    except Exception as e:
        print(f"Error cleaning {file_path}: {e}")
        return False

def main():
    """Clean up all files with unused imports."""
    files_to_clean = [
        'app/components/EnhancedLoadingStates.tsx',
        'app/components/FuturisticText.tsx',
        'app/components/Icons.tsx',
        'app/components/LoadingStates.tsx',
        'app/components/PerformanceDashboard.tsx',
        'app/components/PerformanceEnhancer.tsx',
        'app/components/SearchBar.tsx',
        'app/components/SearchModal.tsx',
        'app/components/SecurityEnhancer.tsx',
        'app/components/SystemMonitor.tsx',
        'app/components/ThemeToggle.tsx',
        'app/components/UserExperienceEnhancer.tsx'
    ]
    
    for file_path in files_to_clean:
        if os.path.exists(file_path):
            cleanup_file(file_path)

if __name__ == "__main__":
    main()